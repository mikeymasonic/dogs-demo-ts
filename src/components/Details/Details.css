.imageGrid img {
  max-width: 20rem;
  margin: 1rem;
}
