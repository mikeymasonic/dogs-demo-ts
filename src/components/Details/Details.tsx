import React, { useState } from 'react';
import './Details.css';

const Details = ({ breed }: { breed: string }) => {
  const [imageLoaded, setImageLoaded] = useState(false);

  return (
    <section className={'imageGrid'}>
      <img src={breed} alt={breed} onLoad={() => setImageLoaded(true)} />
      {!imageLoaded && <h2>loading image...</h2>}
    </section>
  );
};

export default Details;
