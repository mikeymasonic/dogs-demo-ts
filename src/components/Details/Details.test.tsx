import React from 'react';
import { shallow } from 'enzyme';
import Details from './Details';

describe('Details component', () => {
  it('renders Details', () => {
    const wrapper = shallow(
      <Details breed="https://images.dog.ceo/breeds/hound-afghan/n02088094_1003.jpg" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
