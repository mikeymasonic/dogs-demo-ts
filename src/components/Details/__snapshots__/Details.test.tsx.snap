// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders Details 1`] = `ShallowWrapper {}`;
